typings/index.d.ts

```diff
@@ -0,0 +1,93 @@
+declare module 'reduxful' {
+  // fetchUtils
+  export interface RequestAdapterOptions {
+    method: string;
+    url: string;
+    headers: { [key: string]: string };
+    withCredentials: boolean;
+    body: any;
+  }
+
+  export type RequestAdapter = (options: RequestAdapterOptions) => Promise<any>;
+
+  export type Fetcher = (url: string, outOpts: Object) => Promise<any>;
+
+  export function makeFetchAdapter(
+    fetcher: Fetcher,
+    defaultOptions?: Object
+  ): RequestAdapter;
+
+  // utils
+  export interface Resource {
+    isLoaded?: boolean;
+    isUpdating?: boolean;
+    hasError?: boolean;
+  }
+
+  export function isLoaded(resource?: Resource): boolean;
+
+  export function isUpdating(resource?: Resource): boolean;
+
+  export function hasError(resource?: Resource): boolean;
+
+  export function getResourceKey(
+    reqName: string,
+    params?: { [key: string]: string | number }
+  ): string;
+
+  // reduxful
+  export type ApiDescription = Object;
+
+  export type OptionsFn = (getState: () => any) => Object;
+
+  export interface ApiConfig {
+    requestAdapter?: RequestAdapter;
+    options?: OptionsFn;
```

Comment:
Doesn't this take either a settings object or an options-generating function? Could be:

```typescript
type OptionsFn = (getState: () => any) => Object;

type AdapterConfig<AdapterOpts = void> = {
  requestAdapter?: RequestAdapter;
  options?: AdapterOpts | ((getState: () => any)) => AdapterOpts
}
```

Reply:
Yeap I missed options object 👍

Quick question I like the use of generic to specify custom Opts value, but would there be a point of adding `= void`? - if we use `void` user forgets to add type, options can only become `undefined` type or OptionsFn that does not return anything..

How about

1. Remove default type for generic to enforce Options type (since options field is optional, this may not be ideal) 
2. Set the default generic type to be `object` to be safe?